Pas 41 WIP for login page container

Samantha-norrie opened the pull request:
Hi all, I believe that I have finished up most of the container for the login page, but I am still stuck on centering the text within the login button, coloring the input labels both blue and red within the same component, how to add text within the input boxes, as well as adding Montserrat (I know I already asked about this, sorry for having to ask again)! Could I please get your help on how I should complete these 3 remaining tasks? I'll clean up my css more before I make my final push :)

edit: I noticed that leaving blank links caused my build to fail, I'll change them when I push up the next commit as well
edit2: I seem to have used both tabs and spaces... I'll fix this and push back up again tomorrow

Commit:
PAS-41 created container for login page (WIP)

Commit:
Merge branch 'PAS-41' of https://github.com/VikeLabs/Passr into PAS-41

polygonart6777 commented on src/App.tsx:
```diff
@@ -1,25 +1,11 @@
 import React from 'react';
 import logo from './logo.svg';
```
The logo import may be removed, it doesn't seem to be used.

polygonart6777 commented on src/LoginPage.tsx:
```diff
@@ -0,0 +1,36 @@
+      </a>
+			<a
+        className="New-user-link"
+        href=""
```
Your build will pass if you put in a dummy url, you can use ` href="https://reactjs.org"` for now if you want to.

Samantha-norrie replied:
Thank you! I'll make sure to add those when I next push up!

polygonart6777 reviewed (commented):
The login page looks good! Nice work.
You're doing quite a bit of work in this ticket, it may be a good idea to tackle the form field and button components in separate tickets.

polygonart6777 commented on src/LoginPage.css:
```diff
@@ -0,0 +1,200 @@
+	
+	display: flex;
+	align-items: center;
+	text-align: center;
```
In order to centre the text within the button and also use flex box, you can remove `text-align: center` and replace with `justify-content:center`.  See [css flex](https://css-tricks.com/snippets/css/a-guide-to-flexbox/).   Otherwise, you can also remove `display:flex` and `align-items:center` to centre align the text.

polygonart6777 reviewed (commented):
You can import the Montserrat font by adding `<link href="https://fonts.googleapis.com/css2?family=Montserrat&display=swap" rel="stylesheet">` to the index.tsx file inside the `<React.StrictMode>` tag.  Then adding the Montserrat font to both font families `Index.css`.  See [Add-fonts-to-react-project](https://scotch.io/@micwanyoike/how-to-add-fonts-to-a-react-project)
Alternatively, you can let npm install things for you with `npm install fontsource-montserrat` see installation instructions [here](https://www.npmjs.com/package/fontsource-montserrat).

Commit:
Added Montserrat, centered button text, and added links to prevent build error

Samantha-norrie committed:
Merge branch 'master' into PAS-41

BraidenCutforth commented:
> You can import the Montserrat font by adding `<link href="https://fonts.googleapis.com/css2?family=Montserrat&display=swap" rel="stylesheet">` to the index.tsx file inside the `<React.StrictMode>` tag. Then adding the Montserrat font to both font families `Index.css`. See [Add-fonts-to-react-project](https://scotch.io/@micwanyoike/how-to-add-fonts-to-a-react-project)
> Alternatively, you can let npm install things for you with `npm install fontsource-montserrat` see installation instructions [here](https://www.npmjs.com/package/fontsource-montserrat).

If we are going to add the `<link href="https://fonts.googleapis.com/css2?family=Montserrat&display=swap" rel="stylesheet"/>` to the `public/index.html` inside of the `<head>` tag. Anywhere should be fine. This will add this to the pure html file of the application and this allows it to be loaded faster.

BraidenCutforth commented on src/App.tsx:
```diff
@@ -1,26 +1,11 @@
 import React from 'react';
```
I think we should actually revert this change before we merge the ticket. Another ticket will be responsible for actually organizing our different pages using react-router.

The easiest way to do this, again as the last step of your ticket is to call `git checkout origin/master src/App.tsx`

BraidenCutforth commented on src/index.css:
```diff
@@ -1,13 +1,13 @@
 
 code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+  font-family: source-code-pro, Menlo, Monaco, Montserrat, Consolas, 'Courier New',
```
We can drop Montserrat from this because it is probably not a good code font, unless it is a monospace font.

BraidenCutforth commented on src/index.tsx:
```diff
@@ -4,11 +4,13 @@ import './index.css';
-	document.getElementById('root')
+  <React.StrictMode>
+    <App />
+    <link href="https://fonts.googleapis.com/css2?family=Montserrat&display=swap" rel="stylesheet"/>
```
Let's move this to the `public/index.html` inside the `<head>` tag.

polygonart6777 commented:
So it looks like the only reason your checks are not passing is because there is a mix of tabs and spaces within the index.html file.  If you run `npm run prettier`  (and maybe also `npm run lint:fix`) these should all be fixed up.   One way to check your build and lint locally is to run `npm run build` and `npm run lint`, this will tell you if you have any errors before you push up.

Commit:
PAS-41 styled components commit part 1/2

Commit:
Merge branch 'PAS-41' of https://github.com/VikeLabs/Passr into PAS-41

Samantha-norrie commented:
I am in the process of linting and deleting the css file I created, my most recent commit was done only to save some of my work :)

Commit:
PAS-41 linted changes + Montserrat changes

Commit:
PAS-41 removed css file for LoginPage

Commit:
comments + re-installed styled-components

Samantha-norrie commented:
I am still trying to figure out why I am suddenly having issues with having styled-components installed  :/ I'll revert App.tsx after I solve this issue. I tried to clean up the attributes in my styled-components so that they weren't repeated too much but has some issues with using different scopes. Please let me know if you would like me to change them!

Commit:
PAS-41 add package.json

Commit:
PAS-41 forgot the lock!

Commit:
PAS-41 revert App.tsx

Samantha-norrie reviewed (commented):
Reverted App.tsx back to its original state :) Please let me know if you'd like me to change anything else

Samantha-norrie reviewed (commented):
I just 'resolved' some of the changes that I (think I) have fixed in newer commits :)

BraidenCutforth commented on public/index.html:
```diff
@@ -24,12 +22,14 @@
-    <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
-    <!--
+  <link href="https://fonts.googleapis.com/css2?family=Montserrat&display=swap" rel="stylesheet" />
```
Nice! I noticed though that the styling changed from 4 spaces to 2. @polygonart6777 do you know why prettier rules aren't being applied to HTML files?

BraidenCutforth commented on public/index.html:
```diff
@@ -39,5 +39,6 @@
-</html>
+</body>
+
+</html>
```
Let's add a new line to this file. You can configure your editor to automatically adda newline at the end of a file and remove extras.

BraidenCutforth commented on src/LoginPage.tsx:
```diff
@@ -0,0 +1,174 @@
+import logo from './logo.svg';
+import styled from 'styled-components';
+
+//Styling for entire login container
```
So I know school drills it into you "Comment your code!", "Everyone comments code at work, learn to do it early", and the famous "No one will understand your code if you don't comment it". These are terrible ideologies that many people that do not program for anything except for academia will say. 

The truth is, you want to write readable code. Code should be simple enough in small pieces to understand. Sometimes, we do complicated things, that aren't necessarily intuitive at first glance. Or sometimes we write solutions that are a bit hacky for things beyond our control (A piece of code that has grown overtime and is a mess for example, but you don't have time to rewrite it yet.).

Those sometimes situations are the cases where we sometimes will use comments. 

For the styles, you have written great readable code, with amazing variable names that are descriptive to what the component is. For that reason, I think we can remove these comments, as they don't add new information to the code, and will only likely become outdated when someone updates the code but not the comment.

BraidenCutforth commented on src/LoginPage.tsx:
```diff
@@ -0,0 +1,174 @@
+function LoginPage() {
+	return (
+		<LoginPageContainer>
+			<div>
```
You should be able to get rid of this `div` element so the flexbox positioning from the LoginPageContainer (which is a div with styles) is applied to all of the contained items.

BraidenCutforth commented on src/LoginPage.tsx:
```diff
@@ -0,0 +1,174 @@
+
+//Styling for Passr logo placement
+const PassrLogo = styled.div`
+	position: absolute;
```
I think we should remove absolute positioning. I am assuming you ended up having to do this because of the extra div element mentioned below. Or you may have taken these styles from Figma, which aren't great on their own. I can also help you get some of the styling fixed up.

BraidenCutforth commented on src/LoginPage.tsx:
```diff
@@ -0,0 +1,174 @@
+
+//Styling for login button placement
+const LoginButton = styled.div`
+	position: absolute;
```
Again, absolute positioning. I think we will be able to use something relative instead which will resize better with different screen sizes.

BraidenCutforth commented on src/LoginPage.tsx:
```diff
@@ -0,0 +1,174 @@
+				<PassrLogo>
+					<img src={logo} alt="logo" />
+				</PassrLogo>
+				<form>
```
I think we can actually avoid using the form element itself since we may not actually take advantage of the form submit button (we will make our own style since we have the power of js instead of just html). We will need to capture the form inputs as react state. This is a bit involved with react, but I can help you. If you want to start taking a look at the docs, it might help you. https://reactjs.org/docs/hooks-state.html

Commit:
PAS-41 proper indenting and newline added to index.html

Commit:
PAS-41 quick changes

BraidenCutforth committed:
PAS-41: Cleanup Styles and Add State

- Cleaned up styles for login page
- Generalized some styles for alike components
- Add state for email and password
- Add email validation
- Add submit handler

BraidenCutforth committed:
Minor fixes

Commit:
PAS-41 removed useEffect
